eng/pipelines/templates/Build_and_UnitTest.yml

```diff
@@ -466,21 +466,15 @@ steps:
     artifactName: "symbols - $(RtmLabel)"
   condition: " and(succeeded(), eq(variables['IsOfficialBuild'], 'true')) "
 
-- task: ms-vscs-artifact.build-tasks.artifactSymbolTask-1.artifactSymbolTask@0
-  displayName: "Publish Symbols on Symweb"
-  inputs:
-    symbolServiceURI: "https://microsoft.artifacts.visualstudio.com/DefaultCollection"
-    requestName: "CollectionId/$(System.CollectionId)/ProjectId/$(System.TeamProjectId)/$(TeamName)/BuildId/$(Build.BuildId)"
-    sourcePath: "$(Build.Repository.LocalPath)\\artifacts\\symbolstoindex"
-    detailedLog: "true"
-    expirationInDays: "45"
-    usePat: "false"
-  condition: " and(succeeded(),eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true')) "
```

Comment:
@nkolev92 @dtivel or anyone else,  do you know if there's a reason why we've been publishing symbols for 45 days on each build? As part of our insertion process, we already publish symbols (and not limited to 45 days), so I can't think of any reason this step is actually useful.

Reply:
Actually, maybe it's useful for manual vendor testing and OptProf tests, so if VS crashes we have symbols?   Not sure. Although honestly I'm not sure that we'd analyze crash dumps in this scenario. We might just run VS locally (via experimental instance) to reproduce.

What do you think?

Reply:
My guess is that publishing on every build improved the debugging experience for all official builds --- at least for 45 days ---, irrespective of whether or not the build was inserted.  Sometimes people test official builds before they may be inserted.  Unless there's a problem, it seems like a good rule to keep.

This change is at least [3 years old](https://github.com/NuGet/NuGet.Client/commit/7537e3a59c211c4896056a2777dec2a79577b28e#diff-6642474c9c1e8dc7f5c5ed2c99c67b4c086580b80032219592c2df3f44f86026R243).  I don't remember details.

Reply:
The `artifactSymbolTask` task is using a deprecated API and has been outputting warnings telling us (well, the task maintainer) to migrate to another task, but we never noticed since we don't look at the log. I got feedback from someone in the MicroBuild team to use the official AzDO "index source and pusblish symbol" task. but it doesn't have an expiration parameter.

Additionally, this `artifactSymbolTask` task doesn't work on MicroBuild's scale set agent pool that we're required to move to, so there is a problem.

Since there's only a hypothetical (who is debugging official builds not inserted?) benefit and it's less effort to delete than to figure out the repercussions of switching to a different task that doesn't have the same feature, I still propose we delete it.

Reply:
Where do symbols from official NuGet builds are published ? If you remove this task, I believe we will not publish symbols for official builds.

Should we migrate to [AzDO task to publish symbols](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/index-sources-publish-symbols?view=azure-devops), as [documented here](https://www.1eswiki.com/wiki/Using_Azure_DevOps_Symbols#Publishing_Symbols)?

Reply:
Our pipeline to insert NuGet into VS publishes symbols. As you can see from the task that I deleted here, it was only publishing for 45 days, which would not be suitable for builds inserted into VS or the .NET SDK anyway.

Reply:
Ah, the 1ES docs say something that the AzDO docs don't (maybe because it's specific to microsoft.artifacts.visualstudio.com?). Publishing there does indeed have an expiration date, so it is directly equivalent to  the old, unmainted, task. I'll update and run another build to test.

Reply:
Reading the docs more carefully, this is something that will be easier in the pipeline that runs after the build, rather than during the build. So again, I propose cutting it out of the build, as this PR is currently doing.